### PR TITLE
[CT-1160] add relevant protos to dydxprotocol

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `AuthenticatorData` message for enhanced account management.
	- Added a querying service for account authenticators, enabling retrieval of specific and all authenticators associated with an account.
	- Implemented account management functionalities for adding and removing authenticators.

- **Enhancements**
	- Expanded data structures for better organization and management of account authenticators.
	- Established a standardized interface for managing and validating accounts with various authenticator types.

### CODE REVIEW

File: proto/dydxprotocol/accountplus/genesis.proto [hunk 1]
```diff
@@ -3,10 +3,32 @@ package dydxprotocol.accountplus;
...
+  // authenticators are the account's authenticators, these can be multiple
+  // types including SignatureVerification, AllOfs, CosmWasmAuthenticators, etc
+  repeated AccountAuthenticator authenticators = 2
+      [ (gogoproto.nullable) = false ];
+}
+
 // Module genesis state
 message GenesisState {
   repeated AccountState accounts = 1 [ (gogoproto.nullable) = false ];
+
```
Comment (jayy04) [inline proto/dydxprotocol/accountplus/genesis.proto:26]: Note that here I chose not to include the params ([link](https://github.com/osmosis-labs/osmosis/blob/dd225bc392ce72939003aa866f813a8c6d96a16f/proto/osmosis/smartaccount/v1beta1/params.proto)) for simplicity. 

not sure if we ever need to make this feature controllable by gov so omitting for now

File: proto/dydxprotocol/accountplus/models.proto [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  // This type information is essential for differentiating authenticators
+  // and ensuring precise data retrieval from the storage layer.
+  string type = 2;
+
+  // Config is a versatile field used in conjunction with the specific type of
+  // account authenticator to facilitate complex authentication processes.
+  // The interpretation of this field is overloaded, enabling multiple
+  // authenticators to utilize it for their respective purposes.
+  bytes config = 3;
+}
```
Comment (teddyding) [inline proto/dydxprotocol/accountplus/models.proto:23]: Nit: new line

File: proto/dydxprotocol/accountplus/models.proto [hunk 3]
```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
```
Comment (teddyding) [inline proto/dydxprotocol/accountplus/models.proto:1]: For my understanding, are these essentially copies from Osmosi protos? Did you make any changes?

File: proto/dydxprotocol/accountplus/genesis.proto (hunk 1 above)
Comment (jayy04) [reply proto/dydxprotocol/accountplus/genesis.proto:27]: added!

osmosis actually also has a list of addresses that can bypass gov proposals ([link](https://github.com/osmosis-labs/osmosis/blob/362a01b08cd27d8e22e6c733a6f2c8c75b1d34df/proto/osmosis/smartaccount/v1beta1/params.proto#L22-L25)), omitted for now

File: proto/dydxprotocol/accountplus/models.proto (hunk 3 above)
Comment (jayy04) [reply proto/dydxprotocol/accountplus/models.proto:1]: yup - mostly just copies from the Osmosis protos

for params, omitted [maximum_unauthenticated_gas](https://github.com/osmosis-labs/osmosis/blob/362a01b08cd27d8e22e6c733a6f2c8c75b1d34df/proto/osmosis/smartaccount/v1beta1/params.proto#L10-L14) and [circuit_breaker_controllers](https://github.com/osmosis-labs/osmosis/blob/362a01b08cd27d8e22e6c733a6f2c8c75b1d34df/proto/osmosis/smartaccount/v1beta1/params.proto#L22-L25)

**Review by teddyding — APPROVED** [review]

**Pull request merged**